src/components/negotiations-panel.tsx

```diff
@@ -0,0 +1,256 @@
+import TreeView from '@material-ui/lab/TreeView';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import TreeItem from '@material-ui/lab/TreeItem';
+import { SpotlightValue } from './spotlight-value';
+import * as NegotiationStatusManager from '../negotiation-status-manager';
+import {
+  RobotTrajectoryManager,
+  NegotiationTrajectoryResponse
+} from '../robot-trajectory-manager';
+
+const useStyles = makeStyles({
+  root: {
+    height: 240,
+    flexGrow: 1,
+    maxWidth: 400,
+  },
+  finished: {
+    backgroundColor: "lightgreen"
+  },
+  rejected: {
+    backgroundColor: "orange"
+  },
+  forfeited: {
+    backgroundColor: "orange"
+  },
+  defunct: {
+    backgroundColor: "red"
+  },
+  unresolved: {
+    backgroundColor: "red"
+  },
+  ongoing: {
+    backgroundColor: "yellow"
+  }
+});
+
+interface Parameter
+{
+  conflict_version : number;
+  sequence : number[];
+};
+
+export interface NegotiationsPanelProps {
+  conflicts : Readonly<Record<string, NegotiationStatusManager.NegotiationConflict>>;
+  spotlight?: Readonly<SpotlightValue<string>>;
+  trajManager?: Readonly<RobotTrajectoryManager>;
+  negotiationTrajStore : Record<string, NegotiationTrajectoryResponse>;
+}
+
+export default function NegotiationsPanel(props: NegotiationsPanelProps): JSX.Element {
+  const { spotlight } = props;
+
+  React.useEffect(() => {
+    if (!spotlight) {
+      return;
+    }
+    // TODO: spotlight
+  }, [spotlight]);
+  
+  /**  Utility conversion functions **/
+  const classes = useStyles();
+
+  const determineStatusText = (status : NegotiationStatusManager.NegotiationStatus,
+    parent_resolved : NegotiationStatusManager.ResolveState) => {
+    let status_text = "";
+    if (status.forfeited)
+      status_text += "  [FORFEITED]";
+    else if (status.rejected)
+      status_text += "  [REJECTED]";
+    else if (status.defunct)
+      status_text += "  [DEFUNCT]";
+    else if (parent_resolved === NegotiationStatusManager.ResolveState.RESOLVED)
+      status_text += "  [FINISHED]";
+    else
+      status_text += "  [ONGOING]";
+    return status_text;
+  };
+  const determineStyle = (status : NegotiationStatusManager.NegotiationStatus, 
+    parent_resolved : NegotiationStatusManager.ResolveState) => {
+    var style = classes.ongoing;
+    if (status.forfeited)
+      style = classes.forfeited;
+    else if (status.rejected)
+      style = classes.rejected;
+    else if (status.defunct)
+      style = classes.defunct;
+    else if (parent_resolved === NegotiationStatusManager.ResolveState.RESOLVED)
+      style = classes.finished;
+    return style;
+  };
+
+  // keep track of parameters so we can send them as requests
+  var nodeid_to_parameters = new Map<string, Parameter>();
+
+  /** Render Negotiation panel contents **/
+  const renderNegotiations = (version : string, conflict : NegotiationStatusManager.NegotiationConflict) => {
```

Comment:
from what i'm reading about Components, you're saying to encapsulate this function and associated callbacks/text conversion functions into a Component? I'm not sure what benefits we can get out of this since it's used in only 1 spot

Reply:
> from what i'm reading about Components, you're saying to encapsulate this function and associated callbacks/text conversion functions into a Component? I'm not sure what benefits we can get out of this since it's used in only 1 spot

Yeah, I was thinking that it'll be easier to test smaller components mocking their props. But it should work either way. Don't worry.